src/transformers/models/llava_next/modeling_llava_next.py

```diff
@@ -352,28 +405,163 @@ def resize_token_embeddings(self, new_num_tokens: Optional[int] = None, pad_to_m
+        labels=None,
+        image_token_index=None,
+        ignore_index=-100,
+    ):
+        """
+        Merge input_ids with with image features into final embeddings
+
+        Args:
+            image_features (`torch.Tensor` of shape `(all_feature_lens, embed_dim)`)
+                All vision vectors of all images in the batch
+            feature_lens (`torch.LongTensor` of shape `(num_images)`)
+                The length of visual embeddings of each image as stacked in `image_features`
+            inputs_embeds (`torch.Tensor` of shape `(batch_size, sequence_length, embed_dim)`)
+                token embeddings before merging with visual embeddings
+            input_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`)
+                input_ids of tokens, possibly filled with image token
+            attention_mask (`torch.LongTensor` of shape `(batch_size, sequence_length)`)
+                attention_mask for input_ids
+            position_ids (`torch.LongTensor` of shape `(batch_size, sequence_length)`)
+            labels (`torch.Tensor` of shape `(batch_size, sequence_length)`, *optional*, defaults to None)
+                labels need to be recalculated to support training (if provided)
+        Returns:
+            final_embedding, final_attention_mask, position_ids, final_labels
+
+        Explanation:
+            each image has variable length embeddings, with length specified by feature_lens
+            image_features is concatenation of all visual embed vectors
+            task: fill each <image> with the correct number of visual embeddings
+            Example:
+                X (5 patches), Y (3 patches), Z (8)
+                X, Y is on the same sequence (in-context learning)
+            if right padding
+                input_ids: [
+                    a b c d e f X g h i j k Y l m
+                    o p q r Z s t u v _ _ _ _ _ _
+                ]
+                input_ids should be: [
+                    a b c d e f X X X X X g h i j k Y Y Y l m
+                    o p q r Z Z Z Z Z Z Z Z s t u v _ _ _ _ _
+                ]
+                labels should be: [
+                    a b c d e f _ _ _ _ _ g h i j k _ _ _ l m
+                    o p q r _ _ _ _ _ _ _ _ s t u v _ _ _ _ _
+                ]
+            elif left padding
+                input_ids: [
+                    a b c d e f X g h i j k Y l m
+                    _ _ _ _ _ _ o p q r Z s t u v
+                ]
+                input_ids should be: [
+                    a b c d e f X X X X X g h i j k Y Y Y l m
+                    _ _ _ _ _ o p q r Z Z Z Z Z Z Z Z s t u v
+                ]
+                labels should be: [
+                    a b c d e f _ _ _ _ _ g h i j k _ _ _ l m
+                    _ _ _ _ _ o p q r _ _ _ _ _ _ _ _ s t u v
+                ]
+            Edge cases:
+                * If tokens are same but image token sizes are different, then cannot infer left or right padding
```

Comment:
hmmm, this is tricky. I'd be tempted to default to padding_side left if in doubt (with possible warning), as this is typically wanted when generating. WDYT @ArthurZucker 

update: I can see this default is what you did :) I don't think we should have the model attribute `self.padding_side` however

Reply:
@amyeroberts There is no way around this problem, so you can only decide how to handle padding_side based on your conventions, either assumption, warning, or raise error...

Reply:
@NielsRogge @amyeroberts The only way to solve this is bring this whole calculation of `image_to_overwrite` and `text_to_overwrite` to the processor, where it knows what the `padding_side` is, and feed `image_to_overwrite`, `text_to_overwrite` and extended `final_attention_mask` to `model.forward` (and you need to handle batching for them, and it will got worse in distributed training).

https://github.com/huggingface/transformers/blob/5ee0f0f4ef2527ddbc98313070909bcc8ac12565/src/transformers/models/llava_next/modeling_llava_next.py#L564

And for that, you need a new PR and big change to the whole llava_next implementation.

Reply:
Agreed on moving the whole calculation to processing. I believe this is something being discussed already. That would make handling of generation for these models a lot easier

Reply:
For bring the calculation to processor, let's do that in another PR, due to too much changes.